dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/LuceneQueryDateTimeFormatter.java

```diff
@@ -0,0 +1,253 @@
+package com.dotcms.content.elasticsearch.business;
+
+import static com.dotcms.content.elasticsearch.business.ESMappingAPIImpl.dateFormat;
+import static com.dotcms.content.elasticsearch.business.ESMappingAPIImpl.datetimeFormat;
+
+import com.dotmarketing.business.CacheLocator;
+import com.dotmarketing.cache.FieldsCache;
+import com.dotmarketing.portlets.structure.model.Field;
+import com.dotmarketing.portlets.structure.model.Structure;
+import com.dotmarketing.util.DateUtil;
+import com.dotmarketing.util.RegEX;
+import com.dotmarketing.util.RegExMatch;
+import com.dotmarketing.util.UtilMethods;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+
+
+/**
+ *
+ * @author nollymar
+ */
+public class LuceneQueryDateTimeFormatter {
+
+
+    //Date formats supported by dotCMS in lucene queries
+    private static final Map<String, String> LUCENE_DATE_TIME_FORMAT_PATTERN = new LinkedHashMap<String, String>(){
```

Comment:
PATTERN in the name should be plural "PATTERNS" like the next map

Reply:
Done